main/src/ca/uwaterloo/flix/language/phase/Desugar.scala

```diff
@@ -555,6 +555,15 @@ object Desugar {
       val e2 = visitExp(exp2)
       Expr.LetRec(ident, ann, mod, e1, e2, loc)
 
+    case WeededAst.Expr.LocalDef(ident, fparams, dtpe, deff, exp1, exp2, loc) =>
+      val fps = visitFormalParams(fparams)
+      val t = dtpe.map(visitType)
+      val ef = deff.map(visitType)
+      val e1 = visitExp(exp1)
+      val e11 = Expr.Ascribe(e1, t, ef, e1.loc)
```

Comment:
I think there's an invariant that t and ef cannot both be None (at least Let checks so i think you should too)

Reply:
Yes 👍

Reply:
We should just do the same as we do for let, whatever that is.